File: assertion-roulette/java-junit/Should_collect_data_sets_from_tags.java
import org.junit.Test;
import static org.junit.Assert.*;

public class SpecificationProviderTests {

    @Test
    public void shouldCollectDataSetsFromTags() {
        {
            var sut = createSut();
            var result = sut.getSpecification(new Tag[] { new Tag(null, "@DataSource:path\\to\\file.csv"), new Tag(null, "@DataSet:data-set-name") }, SOURCE_FILE_PATH);
            assertNotNull(result);
            assertNotNull(result.getDataSet());
            assertEquals("data-set-name", result.getDataSet());
        }
    }
}

File: smelly/assertion-roulette/CSharpXUnit/CSharpXUnit.Tests/Tests/Should_collect_data_sets_from_tags.cs
using Xunit;

namespace DefaultNamespace
{
    public class SpecificationProviderTests
    {
        [Fact]
        public void Should_collect_data_sets_from_tags()
        {
            {
                var sut = CreateSut();
                var result = sut.GetSpecification(new[] { new Tag(null, @"@DataSource:path\to\file.csv"), new Tag(null, @"@DataSet:data-set-name"), }, SOURCE_FILE_PATH);
                Assert.NotNull(result);
                Assert.NotNull(result.DataSet);
                Assert.Equal("data-set-name", result.DataSet);
            }
        }
    }
}
